aostypes/aostypes.go

```diff
@@ -19,13 +19,33 @@ package aostypes
 
 import (
 	"encoding/json"
+	"math"
+	"regexp"
 	"strconv"
 	"strings"
 	"time"
 
 	"github.com/aoscloud/aos_common/aoserrors"
 )
 
+/***********************************************************************************************************************
+ * Consts
+ **********************************************************************************************************************/
+
+//nolint
+const (
+	alternativePattern = `^P(((?P<year>\d+)-)((?P<month>\d+)-)((?P<day>\d+)))?(T((?P<hour>\d+):)((?P<minute>\d+):)(?P<second>\d+))?$`
+	canonicPattern     = `^P((?P<year>\d+)Y)?((?P<month>\d+)M)?((?P<week>\d+)W)?((?P<day>\d+)D)?(T((?P<hour>\d+)H)?((?P<minute>\d+)M)?((?P<second>\d+)S)?)?$`
+)
+
+// nolint
+var (
+	DayDuration   = 24 * time.Hour
```

Comment:
1. Make these vars private;
2. Year duration 365.25 * DayDuration;
3. MonthDuration 365.25 / 12 * DayDuration or simple YearDuration / 12;
Also, I prefer to have them as const not var. It could be done as the following:
```go
const (
	dayDuration   = 24 * time.Hour
	weekDuration  = 7 * dayDuration
	monthDuration = yearDuration / 12
	yearDuration  = 365*dayDuration + 6*time.Hour
)
```